index.js

```diff
@@ -47,51 +47,247 @@ const player = {
+
+function convertSecToMinFormat(sec) {
+  // sec => mm:ss format.
+  const holeMin = Math.floor(sec / 60)
+  const secondsLeft = sec % 60
+  return formatNumber(holeMin) + ':' + formatNumber(secondsLeft)
+}
+
+function convertMinFormatToSec(minFormat) {
+  const arr = minFormat.split(':').map((str) => Number(str))
+  return 60 * arr[0] + arr[1]
+}
+
+function formatNumber(num) {
+  // to format from "6" => "06".
+  // note: only works up to 99.
+  return ('0' + num).slice(-2)
+}
+
+// ------------------
+
+function getSong(id) {
+  return getObjectFromArray(id, player.songs)
+}
+
+function getPlaylist(id) {
+  return getObjectFromArray(id, player.playlists)
+}
+
+function getObjectFromArray(objId, objectArr) {
+  // throws Error if Object does not exists.
+  const requestedObject = objectArr.find((obj) => obj.id === objId)
+  if (requestedObject === undefined) throw new Error('so such object exists')
+  return requestedObject
+}
+
+function removeSongFromSongs(id) {
+  removeObjectFromArray(id, player.songs)
+}
+
+function removeObjectFromArray(objId, objectArr) {
+  objectArr.splice(objectArr.indexOf(getObjectFromArray(objId, objectArr)), 1)
+}
+
+function removeSongFromPlaylists(id) {
+  let indexOfSong
+  for (let playlist of player.playlists) {
+    // Remember: .indexOf(element) returns -1 if the element is not present in the array.
+    if ((indexOfSong = playlist.songs.indexOf(id)) !== -1) {
+      playlist.songs.splice(indexOfSong, 1)
+    }
+  }
+}
+
+function generateNewIdInArrayOfObjects(objectArr) {
+  /* Gets the largest id and returns it + 1. 
+  i.e the returning id does not exists in the
+  array and is unique.
+  */
+  const idArray = getIdsArrayFromObjArray(objectArr)
+  if (idArray.length === 0) return 1
+  return Math.max(...idArray) + 1
+}
+
+function getIdsArrayFromObjArray(objectArr) {
+  return objectArr.map((obj) => obj.id)
+}
+
+function checkIfObjectIdTaken(id, getFunction) {
+  // getFunction is a function expression. we will know where to search depending on the function
+  try {
+    getFunction(id) // remember: getFunction can be: [getSong or getPlaylist] and throws Error if no song was found.
+    throw new Error('id already exist')
+  } catch (error) {
+    /* 
+    Only if the error is because of "finding" the song
+    then we continue and throw an Error.
+    */
+    if (error.message === 'id already exist') {
+      throw new Error('id already exist')
+    }
+  }
+}
+
+const durationReducer = (totalDuration, songId) => totalDuration + getSong(songId).duration
+
+function searchByQueryInSongs(query) {
+  return searchByQueryInArr(query, player.songs).sort(compareFunction('title'))
+}
+
+function searchByQueryInPlaylist(query) {
+  return searchByQueryInArr(query, player.playlists).sort(compareFunction('name'))
+}
+
+function searchByQueryInArr(query, arr) {
+  let matchArr = []
+  for (let obj of arr) {
+    for (let property of Object.values(obj)) {
+      if (('' + property).toLowerCase().includes(query)) {
```

Comment:
If the `'' + property` is a way to turn things into strings, then you should use the `toString` method. Your code should not only work, but also be readable.